packages/syft/src/syft/core/node/common/client_manager/oblv_api.py

```diff
@@ -0,0 +1,155 @@
+from ...abstract.node import AbstractNodeClient
+from ...enums import RequestAPIFields
+from ...enums import ResponseObjectEnum
+from ..action.exception_action import ExceptionMessage
+from ..node_service.oblv.oblv_messages import CheckEnclaveConnectionMessage
+from ..node_service.oblv.oblv_messages import CreateKeyPairMessage
+from ..node_service.oblv.oblv_messages import DeductBudgetMessage
+from ..node_service.oblv.oblv_messages import GetPublicKeyMessage
+from ..node_service.oblv.oblv_messages import PublishApprovalMessage
+from ..node_service.oblv.oblv_messages import PublishDatasetMessage
+from ..node_service.oblv.oblv_messages import SyftOblvClient
+from .request_api import RequestAPI
+
+
+class OblvAPI(RequestAPI):
+    def __init__(self, client: AbstractNodeClient):
+        super().__init__(
+            client=client,
+            response_key=ResponseObjectEnum.OBLV,
+        )
+
+    def get_key(self, **kwargs: Any) -> Any:
+        response = self.perform_api_request(
+            syft_msg=GetPublicKeyMessage, content=kwargs
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+        else:
+            content = getattr(response, "response")
+            if content is None:
+                raise Exception(f"{type(self)} has no response")
+            return content
+
+    def create_key(self, **kwargs: Any) -> Any:
+        response = self.perform_api_request(
+            syft_msg=CreateKeyPairMessage, content=kwargs
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+        else:
+            content = getattr(response, "resp_msg")
+            if content is None:
+                raise Exception(f"{type(self)} has no response")
+            return content
+
+    def check_connection(self, deployment: DeploymentClient, **kwargs: Any) -> Any:
+        content = {
+            "deployment_id": deployment.deployment_id,
+            "client": SyftOblvClient.from_client(deployment.oblv_client),
+        }
+        response = self.perform_api_request(
+            syft_msg=CheckEnclaveConnectionMessage, content=content
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+        else:
+            content = getattr(response, "resp_msg")
+            if content is None:
+                raise Exception(f"{type(self)} has no response")
+            return content
+
+    def send_dataset(
+        self,
+        deployment: DeploymentClient,
+        dataset: Union[str, TensorWrappedPhiTensorPointer],
+        **kwargs: Any,
+    ) -> Any:
+        content = {
+            "deployment_id": deployment.deployment_id,
+            "client": SyftOblvClient.from_client(deployment.oblv_client),
+        }
+        dataset_id = (
+            dataset if type(dataset) == "str" else dataset.id_at_location.to_string()
```

Comment:
Q: What means it means that the dataset is a str?
MAYBE: Why not `isinstance`?

Reply:
Solved by [930bba3](https://github.com/OpenMined/PySyft/pull/7202/commits/930bba34747b97eec3fef12c388e12db60c7e7f4)